test/e2e/tests/app_test.go

```diff
@@ -48,12 +48,19 @@ func TestApp_Hash(t *testing.T) {
 		info, err := client.ABCIInfo(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, info.Response.LastBlockAppHash, "expected app to return app hash")
+		// In next-block execution, the app hash is stored in the next block
+		block_height := info.Response.LastBlockHeight + 1
 
-		status, err := client.Status(ctx)
-		require.NoError(t, err)
-		require.NotZero(t, status.SyncInfo.LatestBlockHeight)
+		for {
```

Comment:
I'm a little worried about spinning this hot in the test (status isn't going to change until the next block height, which could be a few seconds. 

I think the easiest way to get some kind of intermittent retry is to use `require.Eventually`, but it's possible to write your own retry logic with a timer/etc which might be more clean.

Reply:
I was hesitating what to do here. In the end I decided to leave it as a sort of "active wait" because we are checking the results at every iteration, and so, it might uncover any bad transition in the state (e.g., some race condition)

Reply:
my fear is that it's way more likely to cause some kind of deadlock or create unintentional backpressure, as a result of overtaxing the RPC system that this call relies on.